admin/server/reports.go

```diff
@@ -502,6 +561,186 @@ func (s *Server) generateReportName(ctx context.Context, depl *database.Deployme
+			return nil, status.Error(codes.Internal, err.Error())
+		}
+		projectPermissions, err := s.admin.ProjectPermissionsForUser(ctx, projectID, *ownerID, orgPerms)
+		if err != nil {
+			return nil, status.Error(codes.Internal, err.Error())
+		}
+
+		// Generate JWT attributes based on the creating user's, but with limited project-level permissions.
+		// We store these attributes with the magic token, so it can simulate the creating user (even if the creating user is later deleted or their permissions change).
+		//
+		// NOTE: A problem with this approach is that if we change the built-in format of JWT attributes, these will remain as they were when captured.
+		// NOTE: Another problem is that if the creator is an admin, attrs["admin"] will be true. It shouldn't be a problem today, but could end up leaking some privileges in the future if we're not careful.
+		attrs, err := s.jwtAttributesForUser(ctx, *ownerID, orgID, projectPermissions)
+		if err != nil {
+			return nil, status.Error(codes.Internal, err.Error())
+		}
+		mgcOpts.Attributes = attrs
+	}
+
+	if mv, ok := args["metricsView"]; ok {
+		mgcOpts.MetricsView = mv.(string)
+	} else {
+		return nil, errors.New("missing metricsView in query args")
+	}
+
+	if filter, ok := args["where"]; ok {
+		filterJSON, err := json.Marshal(filter)
+		if err != nil {
+			return nil, fmt.Errorf("failed to marshal filter JSON: %w", err)
+		}
+		mgcOpts.MetricsViewFilterJSON = string(filterJSON)
+	}
+
+	if dims, ok := args["dimensions"]; ok {
+		dimsMapSlice := dims.([]interface{}) // deliberately panics if not a slice
+		for _, dimMap := range dimsMapSlice {
+			dim := dimMap.(map[string]interface{}) // deliberately panics if not a map
+			mgcOpts.MetricsViewFields = append(mgcOpts.MetricsViewFields, dim["name"].(string))
+		}
+	}
+
+	if measures, ok := args["measures"]; ok {
+		measuresMapSlice := measures.([]interface{}) // deliberately panics if not a slice
+		for _, measureMap := range measuresMapSlice {
+			measure := measureMap.(map[string]interface{}) // deliberately panics if not a map
+			mgcOpts.MetricsViewFields = append(mgcOpts.MetricsViewFields, measure["name"].(string))
+		}
+	}
+
+	if len(mgcOpts.MetricsViewFields) == 0 {
+		return nil, errors.New("missing dimensions and measures in query args")
+	}
+
+	// issue magic tokens for new external emails
+	cctx, tx, err := s.admin.DB.NewTx(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("failed to start transaction: %w", err)
+	}
+	defer func() { _ = tx.Rollback() }()
+
+	for email := range emailSet {
+		if ownerID == nil {
+			// set user attrs as per the email
+			mgcOpts.Attributes = map[string]interface{}{
+				"name":   "",
+				"email":  email,
+				"domain": email[strings.LastIndex(email, "@")+1:],
+				"groups": []string{},
+				"admin":  false,
+			}
+		}
+
+		tkn, err := s.admin.IssueMagicAuthToken(cctx, mgcOpts)
+		if err != nil {
+			return nil, fmt.Errorf("failed to issue magic auth token for email %s: %w", email, err)
+		}
```

Comment:
If we just issue a plain magic auth token, then it will also show up in listings of magic auth tokens in the CLI and the UI. This might be confusing for people who wouldn't know it was created automatically for a report.

Not sure, but here are some ideas for fixing this:
- Having an `internal BOOL` on the magic auth tokens which if set hides it in the magic auth token API responses.
- Turning report tokens into full-blown auth tokens by themselves, separate from magic auth tokens (we already have four token types anyway... 😄)